app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -91,6 +92,19 @@ def save_physical_chassis_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.physical_chassis, hashes, :ems_ref)
   end
 
+  def save_physical_storages_inventory(ems, hashes, target = nil)
+    target = ems if target.nil?
+    deletes = target == ems ? :use_association : []
+
+    hashes.each do |h|
+      h[:physical_rack_id] = h.delete(:physical_rack).try(:[], :id)
+    end
+
+    child_keys = %i(computer_system asset_detail)
+    save_inventory_multi(ems.physical_storages, hashes, deletes, [:ems_ref], child_keys)
```

Comment:
Same here there is no ems.physical_storages yet

Reply:
Fixed it, now it should work.